tripleo-heat-templates-diff/apply_patch.py

```diff
@@ -19,46 +35,67 @@
 VERSION_1_CHECK = "openstack-tripleo-heat-templates-8.0.2-4.el7ost.noarch"
 VERSION_2_CHECK = "openstack-tripleo-heat-templates-8.0.2-43.el7ost.noarch"
 VERSION_3_CHECK = "openstack-tripleo-heat-templates-8.0.4-20.el7ost.noarch"
-
 PRE_VERSION_1_DIFF = "diff_OSPD13_8.0.2-4"
 VERSION_2_DIFF = "diff_OSPD13_8.0.2-43"
 VERSION_3_DIFF = "diff_OSPD13_8.0.4-20"
 
 
-if len(sys.argv) != 1:
-    print "Usage: %s" % sys.argv[0]
-    sys.exit(1)
-
 def version_compare((e1, v1, r1), (e2, v2, r2)):
     return rpm.labelCompare((e1, v1, r1), (e2, v2, r2))
 
-version = subprocess.check_output(['rpm', '-qa', 'openstack-tripleo-heat-templates'])
 
-(e0, v0, r0) = stringToVersion(version)
-(e1, v1, r1) = stringToVersion(VERSION_1_CHECK)
-(e2, v2, r2) = stringToVersion(VERSION_2_CHECK)
-(e3, v3, r3) = stringToVersion(VERSION_3_CHECK)
+def file_exists(filename):
+    if not os.path.exists(filename):
+        print "File: %s is not present in the current directory. " \
+              "Please check!!" % filename
+        sys.exit(1)
+    return filename
+
+
+def main():
+
+    len(sys.argv) != 1:
+        print "Usage: %s" % sys.argv[0]
+        sys.exit(1)
+
+    version = subprocess.check_output(
+        ['rpm', '-qa', 'openstack-tripleo-heat-templates']
+    )
+
+    (e0, v0, r0) = stringToVersion(version)
+    (e1, v1, r1) = stringToVersion(VERSION_1_CHECK)
+    (e2, v2, r2) = stringToVersion(VERSION_2_CHECK)
+    (e3, v3, r3) = stringToVersion(VERSION_3_CHECK)
+
+    args = "patch -p0 -N -d /usr/share"
+
+    # Compare versions
+    version_1_rc = version_compare((e0, v0, r0), (e1, v1, r1))
+    version_2_rc = version_compare((e0, v0, r0), (e2, v2, r2))
+    version_3_rc = version_compare((e0, v0, r0), (e3, v3, r3))
+
+    if version_1_rc <= 0:
+        args = args + " < " + PRE_VERSION_1_DIFF
```

Comment:
I am comparing with OSPD-10 changes, did you forgot to call file_exists here and below as well?